### DISCUSSION TITLE
Handle Vectors

### DISCUSSION THREAD

**Discussion by leo-paz** [planetscale/discussion#526] [Feature Requests] [answered]
I know very little about how PlanetScale works and how feasible this is but just wondering if there were any plans to support vector embeddings? I absolutely love using PlanetScale for other things and need a vector db so was probably going to use one of the existing dbs out there but was wondering if PlanetScale had a plan for this anytime soon?

**Comment by orware** [comment]
Hello @leo-paz,

**Update:** We're [working on it](https://planetscale.com/blog/planetscale-is-bringing-vector-search-and-storage-to-mysql)!

~~Due to the [specialized nature of vector embeddings](https://medium.com/@tenyks_blogger/vector-databases-unlock-the-potential-of-your-data-5bd4950bd72), if a new capability within MySQL were to emerge in the future that helped with those, we would likely be able to provide some support for that functionality in the future, but I wouldn't say that there are any plans to provide anything custom for this need at this time within PlanetScale itself.~~

~~There was a [recent discussion about the topic](https://twitter.com/wesbos/status/1653902175272329217?lang=en) in relation to storing that type of data in MySQL that mainly mentioned using column types such as the `JSON` one as a workable option, but it would likely come with some sort of performance tradeoffs and caveats compared to the purpose-built solutions.~~

~~Going with one of the existing vector databases out there in your case sounds like it could be the best way to go!~~

**Comment by CanRau** [comment]
For reference [PlanetScale is bringing vector search and storage to MySQL](https://planetscale.com/blog/planetscale-is-bringing-vector-search-and-storage-to-mysql)

**Comment by mscoutermarsh** [comment, marked as answer]
Vectors are now available in public beta: https://planetscale.com/blog/announcing-planetscale-vectors-public-beta
